### PR TITLE
Allow admins to copy resources (TAM-121)

### PR DESCRIPTION
**Pull request by urshala**
Allow admins to copy the resources. The copied resource should
have a different name (original name _COPY_count). and set as 
hidden so that it won't be visible in the varaamo until  changed
manually.

resource_copy

### CODE REVIEW

File: locale/sv/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -1563,6 +1566,13 @@ msgstr "Modifiera resurs"
 msgid "Create new resource"
 msgstr "Skapa ny resurs"
 
+#, python-brace-format
+msgid "{source_resource.name} copied successfully."
+msgstr ""
+
+msgid "Something went wrong! Please contact admin."
+msgstr ""
```
Comment (jopesy) [inline locale/sv/LC_MESSAGES/django.po]: "Något gick fel! Kontakta admin."

File: locale/sv/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -1563,6 +1566,13 @@ msgstr "Modifiera resurs"
 msgid "Create new resource"
 msgstr "Skapa ny resurs"
 
+#, python-brace-format
+msgid "{source_resource.name} copied successfully."
+msgstr ""
```
Comment (jopesy) [inline locale/sv/LC_MESSAGES/django.po]: "{source_resource.name} framgångsrikt kopierat."

File: locale/fi/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -1570,6 +1573,13 @@ msgstr "Muokkaa resurssia"
 msgid "Create new resource"
 msgstr "Luo uusi resurssi"
 
+#, python-brace-format
+msgid "{source_resource.name} copied successfully."
+msgstr ""
```
Comment (jopesy) [inline locale/fi/LC_MESSAGES/django.po]: "{source_resource.name} kopioitu onnistuneesti."

File: locale/fi/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -1570,6 +1573,13 @@ msgstr "Muokkaa resurssia"
...
+
+msgid "Something went wrong! Please contact admin."
+msgstr ""
```
Comment (jopesy) [inline locale/fi/LC_MESSAGES/django.po]: "Jokin meni vikaan! Ota yhteyttä ylläpitäjään."

Comment (urshala) [reply locale/fi/LC_MESSAGES/django.po]: Added the missing translations.

File: respa_admin/views/resources.py [hunk 5]
```diff
@@ -467,6 +470,67 @@ def _delete_extra_resource_accessibility(self, resource_accessibility_formset):
...
+                for source_day in source_days:
+                    source_day.pk = None
+                    source_day.period = item
+                    source_day.save()
+
+
+def _get_resource_copy_name(resource):
+    resource_copy_name = f'{resource.name}_COPY_'
+    copied_resource_count = Resource.objects.filter(name__icontains=resource_copy_name).count()
+    next_copy_count = 1 if copied_resource_count == 0 else copied_resource_count + 1
```
Comment (tuomas777) [inline respa_admin/views/resources.py]: Not sure why the `if` is needed, wouldn't just `next_copy_count = copied_resource_count + 1` give the same results? 🙂

File: respa_admin/views/resources.py [hunk 6]
```diff
@@ -467,6 +470,67 @@ def _delete_extra_resource_accessibility(self, resource_accessibility_formset):
...
+    'images',
+    'opening_hours',
+    'periods',
+    'accessibility_summaries',
+)
+
+
+@transaction.atomic
+def copy_resource(request, resource_id):
+    source_resource = Resource.objects.prefetch_related(*RESOURCE_RELATED_FIELDS).get(pk=resource_id)
```
Comment (tuomas777) [inline respa_admin/views/resources.py]: Not a big deal as the resource should not be normally missing, but using [get_object_or_404()](https://docs.djangoproject.com/en/4.0/topics/http/shortcuts/#get-object-or-404) would be a bit nicer to make sure there will be no uncaught exceptions.

**Review by tuomas777 — APPROVED** [review]
Added a couple of minor comments, but in general LGTM, good job! 👍 🚀 🐰

File: respa_admin/views/resources.py (hunk 6 above)
Comment (urshala) [reply respa_admin/views/resources.py]: Made the suggested changes.

File: respa_admin/views/resources.py (hunk 5 above)
Comment (urshala) [reply respa_admin/views/resources.py]: Fixed it .

**Pull request merged**